ovc5/software/ros_drivers/ros1/CMakeLists.txt

```diff
@@ -49,7 +50,7 @@ if (catkin_FOUND)
   catkin_package(
     LIBRARIES ovc_driver
     CATKIN_DEPENDS roscpp sensor_msgs cv_bridge dynamic_reconfigure
-    #DEPENDS system_lib
+    DEPENDS jsoncpp
```

Comment:
I think we can avoid having `jsoncpp` as a dependency to every package (I see that you added it to `libovc`, `camera_host_driver` and the ROS1 driver, probably following the same approach it will also have to be added to the ROS2 driver?).
You could have only the main package that sends the JSON message have it as a dependency then expose APIs that are independent of JSON to the other nodes, i.e. instead of having an `updateConfig` function that takes a `Json::Value` as a parameter, you could have a `setFrameRate` function, a `setExposure(int cam_id, float exposure)` function etc.
This would also make downstream packages a bit simpler because users wouldn't need to learn about the JSON syntax, as well as making maintenance simpler since if we change the format we need to update only a single function in `libovc` instead of chasing down all the client nodes that are building their own JSON message

Reply:
I agree with this but also didn't want to make any assumptions right away. I imagine you would want to update a bunch of parameters simultaneously across multiple cameras (to minimize downtime) so you need to batch all of the commands if you want it in an API format. I was thinking that, for now, JSON works albeit not the best solution. Down the road, we can discuss the best API structure to remove this direct exposure.